x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/indicator_match/threat_mapping/get_event_count.ts

```diff
@@ -40,14 +39,16 @@ export const getEventList = async ({
     `Querying the events items from the index: "${index}" with searchAfter: "${searchAfter}" for up to ${calculatedPerPage} indicator items`
   );
 
-  const queryFilter = getQueryFilter({
-    query,
-    language: language ?? 'kuery',
-    filters,
-    index,
-    exceptionFilter,
-    fields: indexFields,
-  });
+  const queryFilter = await getQueryFilter(
```

Comment:
This call and the one in `getThreatList` are where we are most at risk of `fieldCaps` calls in loops. In `createThreatSignals` we call either `getThreatList` or `getEventList` in a loop, fetching a few thousand docs at a time. We call `getQueryFilter` for each page, but the query doesn't actually change. I think we could precompute the query for `getEventList` [on line 308 here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/indicator_match/threat_mapping/create_threat_signals.ts#L308) and precompute the query for `getThreatList` [on line 376](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/indicator_match/threat_mapping/create_threat_signals.ts#L376).

However, as I look further `getThreatList` also gets called through other convoluted code paths in loops. Perhaps the easiest solution would be to ignore threat match rules for now and focus on limiting the field caps request size for query and EQL rules, as those 2 are by far the most common rules and dominate the data transfer usage.